File: dashboard/src/api/staffIdentity.ts
import { getAPI, postAPI } from "./client";

export interface StaffProfile {
	staff_id: string;
	display_name: string;
	active: boolean;
	created_at: string;
	updated_at: string;
}

export interface StaffProfilesResponse {
	profiles: StaffProfile[];
}

export function fetchStaffProfiles(includeInactive = false): Promise<StaffProfilesResponse> {
	return getAPI<StaffProfilesResponse>(
		`/api/staff/profiles?include_inactive=${includeInactive ? "true" : "false"}`,
	);
}

export function createStaffProfile(display_name: string): Promise<StaffProfile> {
	return postAPI<StaffProfile>("/api/staff/profiles", { display_name });
}

File: agent/models/staff_identity.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class StaffProfileResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    staff_id: str
    display_name: str
    active: bool
    created_at: datetime | str
    updated_at: datetime | str


class StaffProfileCreateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    # Display name is collected only by the profile-management surface. It is
    # never accepted as per-generation attribution.
    display_name: str = Field(min_length=1, max_length=160)


class StaffProfileUpdateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    display_name: str | None = Field(default=None, min_length=1, max_length=160)
    active: bool | None = None


class StaffProfilesResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    profiles: list[StaffProfileResponse]
